extensions/amp-ooyala-player/0.1/amp-ooyala-player.js

```diff
@@ -0,0 +1,210 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {tryParseJson} from '../../../src/json';
+import {user} from '../../../src/log';
+import {removeElement} from '../../../src/dom';
+import {isObject} from '../../../src/types';
+import {VideoEvents} from '../../../src/video-interface';
+import {videoManagerForDoc} from '../../../src/video-manager';
+
+/**
+ * @implements {../../../src/video-interface.VideoInterface}
+ */
+class AmpOoyalaPlayer extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?Element} */
+    this.iframe_ = null;
+
+    /** @private {?Promise} */
+    this.playerReadyPromise_ = null;
+
+    /** @private {?Function} */
+    this.playerReadyResolver_ = null;
+  }
+
+  /**
+   * @param {boolean=} opt_onLayout
+   * @override
+   */
+  preconnectCallback(opt_onLayout) {
+    this.preconnect.url('https://player.ooyala.com', opt_onLayout);
+  }
+
+  /** @override */
+  buildCallback() {
+    this.playerReadyPromise_ = new Promise(resolve => {
+      this.playerReadyResolver_ = resolve;
+    });
+
+    const iframe = this.element.ownerDocument.createElement('iframe');
+    this.iframe_ = iframe;
+
+    this.forwardEvents([VideoEvents.PLAY, VideoEvents.PAUSE], iframe);
+    this.applyFillContent(iframe, true);
+    this.element.appendChild(iframe);
+
+    iframe.setAttribute('frameborder', '0');
+    iframe.setAttribute('allowfullscreen', 'true');
+
+    videoManagerForDoc(this.win.document).register(this);
+  }
+
+  /** @override */
+  layoutCallback() {
+    const embedCode = user().assert(
+      this.element.getAttribute('data-embedcode'),
+      'The data-embedcode attribute is required for <amp-ooyala-player> %s',
+      this.element);
+    const pCode = user().assert(
+      this.element.getAttribute('data-pcode'),
+      'The data-pcode attribute is required for <amp-ooyala-player> %s',
+      this.element);
+    const playerId = user().assert(
+      this.element.getAttribute('data-playerid'),
+      'The data-playerid attribute is required for <amp-ooyala-player> %s',
+      this.element);
+    const autoplay = this.element.getAttribute('autoplay');
+
+    let src = 'https://player.ooyala.com/iframe.html?platform=html5-priority';
+    const playerVersion = this.element.getAttribute('data-playerversion') || '';
+    if (playerVersion.toLowerCase() == 'v4') {
+      src = 'https://player.ooyala.com/static/v4/sandbox/amp_iframe/' +
+        'skin-plugin/amp_iframe.html?pcode=' + encodeURIComponent(pCode);
+      const configUrl = this.element.getAttribute('data-config');
+      if (configUrl) {
+        src += '&options[skin.config]=' + encodeURIComponent(configUrl);
+      }
+    }
+
+    src += '&ec=' + encodeURIComponent(embedCode) +
+      '&pbid=' + encodeURIComponent(playerId);
+    if (autoplay) {
```

Comment:
there should be no need to pass `autoplay` to Ooyala player, please allow AMP to manage autoplay behaviour for consistency with other players. autoplay will be managed automatically using the methods and events you implemented as part of the video-interface.